IronQuotes/src/main/java/com/theironyard/Controllers/IronQuotesController.java

```diff
@@ -0,0 +1,168 @@
+package com.theironyard.Controllers;
+import com.theironyard.Command.QuoteCommand;
+import com.theironyard.Command.TagCommand;
+import com.theironyard.Entities.Quote;
+import com.theironyard.Entities.Tag;
+import com.theironyard.Entities.User;
+import com.theironyard.Services.QuoteRepository;
+import com.theironyard.Services.TagRepository;
+import com.theironyard.Services.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.Valid;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Nigel on 8/1/16.
+ */
+@RestController
+public class IronQuotesController {
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    QuoteRepository quoteRepository;
+
+    @Autowired
+    TagRepository tagRepository;
+
+    /*@RequestMapping(path = "/user", method = RequestMethod.POST)
+    public User createUser(@Valid @RequestBody User user, BindingResult bindingResult,
+                          HttpServletResponse response){
+
+        if(bindingResult.hasErrors()){
+            response.setStatus(400);
+            return null;
+        }
+
+        userRepository.save(user);
+        return user;
+    }*/
+
+
+    //*********CREATE-PATHS*********//
+    @RequestMapping(path = "/quotes", method = RequestMethod.POST)
+    public Quote createQuote(@RequestBody Quote quote){
+        quoteRepository.save(quote);
+
+        return quote;
+    }
+
+    @RequestMapping(path = "/quotes/{id}/tags", method = RequestMethod.POST)
+    public Quote createTag(@PathVariable int id, @RequestBody TagCommand command) throws Exception {
+        Quote quote = quoteRepository.getOne(id);
+
+        if(quote == null){
+            throw new Exception("Quote doesn't exist");
+        }
+
+        Tag tag = tagRepository.findFirstByValue(command.getValue());
+        if(tag == null){
+            tag = new Tag(command.getValue());
+            tagRepository.save(tag);
+        }
+
+        quote.addTag(tag);
+        quoteRepository.save(quote);
+        return quote;
+    }
+
+    //*********READ-PATHS*********//
+    @RequestMapping(path = "/quotes/{id}", method = RequestMethod.GET)
+    public Quote getQuote(@PathVariable int id) throws Exception {
+        Quote quote = quoteRepository.getOne(id);
+
+        if (quote == null){
+            throw new Exception("Quote doesn't exist");
+        }
+
+        return quote;
+    }
+
+    @RequestMapping(path = "/tags/{id}", method = RequestMethod.GET)
+    public Tag getTag(@PathVariable int id) throws Exception {
+        Tag tag = tagRepository.getOne(id);
+
+        if (tag  == null){
+            throw new Exception("Tag doesn't exit");
```

Comment:
This should have been a new exception since this will throw a 500 status on the front end instead of a 400 bad request like it should